builtin/providers/digitalocean/resource_digitalocean_floating_ip.go

```diff
@@ -83,8 +83,10 @@ func resourceDigitalOceanFloatingIpRead(d *schema.ResourceData, meta interface{}
 	}
 
 	if _, ok := d.GetOk("droplet_id"); ok {
-		log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
-		d.Set("region", floatingIp.Droplet.Region.Slug)
+		if floatingIp.Droplet != nil {
+			log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
+			d.Set("region", floatingIp.Droplet.Region.Slug)
+		}
```

Comment:
@franklinhu ok, I just sent #4216. To be more explicit. I don't rely on the d.GetOk anymore. I use the API result directly and then check for a droplet and use the region that way.